src/components/Questionnaire/QuestionTypes/AppointmentQuestion.tsx

```diff
@@ -50,7 +50,10 @@ export function AppointmentQuestion({
     (questionnaireResponse.values?.[0]
       ?.value as unknown as CreateAppointmentQuestion[]) || [];
 
-  const value = values[0] ?? {};
+  const value: CreateAppointmentQuestion = values[0] ?? {
+    reason_for_visit: "",
+    slot_id: "",
```

Comment:
slot id should never be empty string. it can be null/undefined though.

Since this is the question's state, we could use Partial<CreateAppointmentQuestion> instead.